### PR TITLE
Remove Friend Service

### PR DESCRIPTION
**Pull request by umututku03**
## Motivation and Context

This update allows users to promptly remove friends, updating the followed friends list while efficiently removing associated playlists. This streamlining enhances the management of social connections in our music app.

## Your Changes

I've restructured the code in the remove_friend module to adhere to Clean Architecture and SOLID Design Principles. The changes emphasize the isolation of components related to the removal use case, promoting a clear and organized structure. Following the Dependency Inversion Principle, the code now depends on abstractions, enhancing flexibility and maintainability by reducing reliance on high-level modules.

**Description**:

In the aftermath of the update, our database and View States for the user library and followed friends seamlessly align with the prescribed alterations. The removal of a friend is efficiently handled, reflecting in both the data access object and the view state of the followed friends collection. Furthermore, any playlists associated with the removed friend are appropriately expunged from the user's library, maintaining coherence across both the data access object and the view state.

**Type of change** (select all that apply):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Refactoring (internal change to codebase, without changing functionality)
- [ ] Test update (change that modifies or updates tests only)
- [ ] Documentation update (change that modifies or updates documentation only)
- [ ] Other (please specify):

## Testing

I conducted a comprehensive self-review of the code, addressing both back-end and front-end considerations. Notably, I've incorporated unit tests to validate the functionality of the code's back-end. In terms of front-end aspects, I ensured adherence to expected specifications through thorough checks. This dual-focused review aims to enhance the overall robustness and reliability of the codebase.

## Questions and Comments (if applicable)

N/A

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have added tests for my changes. 
- [X] Coverage tests
- [X] Coding style (conforms to Google Java Style Guide)
- [X] Especially follows section 7 (Javadoc is present for every public class, public or protected member)
- [X] Especially follows section 4.6 (Multiple consecutive blank lines are not encouraged)
- [X] Correctness (follows CA and SOLID, is a reasonable and not overly complicated solution, etc.)
- [X] Description of pull request is clear, consise, and reasonably detailed

### CODE REVIEW

File: src/app/Main.java
Comment (ibrahimmmalik) [inline src/app/Main.java:1]: All of the imports on this file can be changed to a single line where just services.remove_friend.* is imported

File: src/services/remove_friend/RemoveFriendPresenter.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  @Override
+  public void prepareSuccessView(RemoveFriendOutputData removeFriendsOutput) {
+    FollowedFriendsViewState followedFriendsViewState = followedFriendsViewModel.state;
+    followedFriendsViewState.friendUsernames = removeFriendsOutput.userFriends;
+    followedFriendsViewModel.firePropertyChanged();
+
+    MyLibraryViewState myLibraryViewState = myLibraryViewModel.state;
+    myLibraryViewState.personalPlaylists = removeFriendsOutput.userPlaylists;
+    followedFriendsViewModel.firePropertyChanged();
```
Comment (HussainE4) [inline src/services/remove_friend/RemoveFriendPresenter.java:28]: This should be firing a property change in the library view model, not the followed friends view model

**Review by HussainE4 — CHANGES_REQUESTED** [review]

File: src/app/Main.java
Comment (ibrahimmmalik) [inline src/app/Main.java:1]: Changes in main are all correct and related to the use case, nothing else was changed that was not necessary. Main follows all SOLID and CA principles. However, just for design, remove_friend import statements can be shortened to 1 line.

File: src/services/remove_friend/RemoveFriendController.java
Comment (ibrahimmmalik) [inline src/services/remove_friend/RemoveFriendController.java:1]: TODOs can be removed from this file now that the task has been completed

**Review by bryjjch — APPROVED** [review]
Great implementation of RemoveFriend service with good formatting in Interactor, Presenter, and Controller. Great implementation of CA and SOLID principles!

**Review by ibrahimmmalik — APPROVED** [review]
Code follows CA and SOLID principles throughout and very clearly only adds the functionality described in the pull request. The naming conventions and formatting of all the files follow the Google Java Style guide as requested in the implementation description.

**Pull request merged**